ivadomed/loader/mri2d_segmentation_dataset.py

```diff
@@ -140,7 +165,19 @@ def prepare_indices(self):
         """Stores coordinates of 2d patches for training."""
         for i in range(0, len(self.handlers)):
 
-            input_img = self.handlers[i][0]['input']
+            if self.disk_cache:
+                with self.handlers[i].open(mode="rb") as f:
+                    item = pickle.load(f)
+                    primary_handle = item[0]
```

Comment:
Here we cache the item "i" of `self.handlers`. However, if we use patching and disk_cache is true, I think this item is already cached at line 142, no?

Reply:
Sorry, I am not sure I follow. These lines are loading the cached data from the disk. So it is the counterpart of the earlier line 142 where write to disk happens. Mainly used to construct indexes and avoid have entire `self.handlers` in memory by only loading the relevant item part one at a time.